xwiki-commons-core/xwiki-commons-xml/src/test/java/org/xwiki/xml/internal/html/DefaultHTMLCleanerTest.java

```diff
@@ -533,25 +535,27 @@ void controlCharacters() throws Exception
         htmlInput = "<p>&#8;</p>";
         document = this.cleaner.clean(new StringReader(htmlInput));
 
-        // HtmlCleaner currently doesn't handle properly unicode characters: asking it to recognize them
-        // involves that all entities will be escaped during the parsing and that's not what we want. So we
-        // keep them encoded.
+        // HtmlCleaner now handles unicode character entities. These are decoded
+        // if possible, and if they are control characters, replaced by whitespace
         // See https://sourceforge.net/p/htmlcleaner/bugs/221/
         textContent =
             document.getElementsByTagName("p").item(0).getTextContent();
-        assertEquals("&#8;", textContent);
-        assertHTML("<p>&#8;</p>", "&#8;");
+        assertEquals(" ", textContent);
+        assertHTML(" ", "&#8;");
```

Comment:
Also new: control characters are now replaced by white space instead of keeping them encoded.

Reply:
I guess it's ok given the first part of the test.